src/lib/create-config.ts

```diff
@@ -669,23 +669,38 @@ export function createConfig(options: IUnleashOptions): IUnleashConfig {
         ),
         constraints: Math.max(
             0,
-            parseEnvVarNumber(process.env.UNLEASH_CONSTRAINTS_LIMIT, 30),
+            parseEnvVarNumber(
+                process.env.UNLEASH_CONSTRAINTS_LIMIT,
+                options?.resourceLimits?.constraints || 30,
+            ),
         ),
-        environments: parseEnvVarNumber(
-            process.env.UNLEASH_ENVIRONMENTS_LIMIT,
-            50,
+        environments: Math.max(
+            0,
+            parseEnvVarNumber(
+                process.env.UNLEASH_ENVIRONMENTS_LIMIT,
+                options?.resourceLimits?.environments || 50,
+            ),
```

Comment:
No, I do think that's what we want. We want to disallow negative values, but we can set it to 0 if you want to.

Reply:
Yeah, we can do that. Been thinking of something like it. I don't think I agree that we should throw an error, though. It feels more like just stopping you from doing something stupid. "Be liberal in what you accept", right?

Reply:
Yes, but if you set the value to zero because they're doing something stupid, the users will not be able to add new environments, and they wouldn't know why. Because the person configuring Unleash is probably a different person than the one using it, I'd like Unleash to fail fast and tell me what I'm doing wrong... if someone sets a negative value, it's likely they're trying to accomplish something... something different than setting the value to zero

Reply:
Actually, we talked about the lower limits with @kwasniew. But I agree, for envs, it should be 1, because you can never have fewer than a single env. 

But effectively it'll stop you from creating new ones, yeah. But again, if you do that, you probably have a reason for it. If it's an issue for the users, they'll get warnings in the UI and can talk to the admin. 

If we think that Unleash should fail at invalid values (like negative limits), that might be a discussion worth having, but I don't think you and I are gonna solve it in this PR. Especially because we seem to disagree about the right course of action here 💁🏼 

Os another point, though, I think we can mention the minimum allowed values in the docs, to make it clearer.